torch_xla/csrc/runtime/runtime.cc

```diff
@@ -34,8 +36,10 @@ ComputationClient* CreateClient() {
 }  // namespace
 
 ComputationClient* GetComputationClient() {
-  std::call_once(g_computation_client_once,
-                 [&]() { g_computation_client = std::move(CreateClient()); });
+  std::call_once(g_computation_client_once, [&]() {
+    g_computation_client = std::move(CreateClient());
+    std::atexit([]() { delete g_computation_client.load(); });
```

Comment:
cc @will-cromar, we need to ensure the ComputationClient is deleted before exiting so that all clients can gracefully shutdown before the DistributedRuntimeService is destroyed.

Reply:
Is it possible to do with from `XLAC._prepare_to_exit`? We need to guarantee that `ComputationClient::PrepareToExit()` is not called after this shutdown hook

Reply:
side note: this looks a lot like just holding a static variable that will be destroyed at the end of the process. I'll look into refactoring how we handle `g_computation_client` after this PR merges.

Reply:
I see, both are registered atexit.. I'll make this a bit more controlled and set g_computation_client to nullptr.

I'm not sure why, but [using PrepareToExit](https://github.com/pytorch/xla/commit/92ba0796fd8402b2d65f177454229d89a1bae737) resulted in segfaults in MP tests: https://github.com/pytorch/xla/actions/runs/6673957659/job/18140734664

Reply:
I suggest changing the pybind to actually delete the client here after calling the client's `PrepareToExit`: https://github.com/pytorch/xla/blob/1f0e972a6dfc5ed3205b23d0fccac8ebb128584f/torch_xla/csrc/init_python_bindings.cpp#L98

Reply:
That's safer, I'll make the change.